src/server/active_tran_server.cpp

```diff
@@ -69,31 +97,38 @@ int active_tran_server::init_page_server_hosts (const char *db_name)
       return ER_HOST_PORT_PARAMETER;
     }
 
-  long port = -1;
-  try
+  size_t pos = 0;
```

Comment:
Nitpicking: move all the code that does page server configuration parsing to a separate function. It could be name `parse_page_server_hosts_config()`.